docs: Add kommander configuration page (#4088)
mesosphere/dcos-docs-site: pages/dkp/kommander/2.1/install/configuration/index.md

---
layout: layout.pug
navigationTitle: Configuration
title: Configuration
menuWeight: 30
excerpt: How to configure a Kommander installation
beta: false
enterprise: false
---

> Kommander can be configured during the initial installation and also post-installation using the Kommander CLI.

Reviewer: ```suggestion
You can configure Kommander during the initial installation, and also post-installation using the Kommander CLI.
```

Author: resolved in a2389daa9fd58b84770bb55c6ee73e7b4e829e54

## Initializing a configuration file

> To get started with configuring Kommander, you can run the following command to initialize a default configuration file:

Reviewer: ```suggestion
To begin configuring Kommander, run the following command to initialize a default configuration file:
```

Author: resolved in a2389daa9fd58b84770bb55c6ee73e7b4e829e54

```bash
kommander install --init > kommander.yaml
```

## Configuring applications

> After you have a default configuration file, you can then configure each `app` either inline **or** by referencing another `yaml` file. The configuration values for each `app` correspond to the Helm Chart values for the application.

Reviewer 2: Do we have somewhere that documents where to find the helm charts?  If so, I would suggest we add a link here.  If not, please consider adding this to this PR.    Thank you!!!!

Author: I didn't find any existing docs, adding a small note in 7f877ee5769731ee35f68e4362c8389ce4f79712. We don't really have a great way to make this discoverable today, the way i've done it is by referencing the HelmRelease for each application.

> Once you have a default configuration file, each `app` can be configured either inline **or** by referencing another `yaml` file. The configuration values for each `app` correspond to the Helm Chart values for the application.

Reviewer: ```suggestion
After you have a default configuration file, you can then configure each `app` either inline **or** by referencing another `yaml` file. The configuration values for each `app` correspond to the Helm Chart values for the application.
```

Author: resolved in a2389daa9fd58b84770bb55c6ee73e7b4e829e54

### Inline configuration (using values)

> In this example, we configure the `centralized-grafana` application with resource limits by defining the Helm Chart values in the Kommander configuration file.

Reviewer: ```suggestion
In this example, you configure the `centralized-grafana` application with resource limits by defining the Helm Chart values in the Kommander configuration file.
```

Author: resolved in a2389daa9fd58b84770bb55c6ee73e7b4e829e54

```yaml
apiVersion: config.kommander.mesosphere.io/v1alpha1
kind: Installation
apps:
  centralized-grafana:
    values: |
      resources:
        limits:
          cpu: 150m
          memory: 100Mi
        requests:
          cpu: 100m
          memory: 50Mi
...
```

### Referencing another yaml file (using valuesFrom)

> Alternatively, we could create another `yaml` file containing the configuration for `centralized-grafana` and reference that using `valuesFrom`. You can point to this file by using either a relative path (from the configuration file location) or by using an absolute path.

Reviewer: ```suggestion
Alternatively, you could create another `yaml` file containing the configuration for `centralized-grafana` and reference that using `valuesFrom`. Point to this file by using either a relative path (from the configuration file location) or by using an absolute path.
```

```yaml
cat > centralized-grafana.yaml <<EOF
grafana:
  resources:
    limits:
      cpu: 150m
      memory: 100Mi
    requests:
      cpu: 100m
      memory: 50Mi
EOF
```

```yaml
apiVersion: config.kommander.mesosphere.io/v1alpha1
kind: Installation
apps:
  centralized-grafana:
    valuesFrom: centralized-grafana.yaml
...
```

## Install with configuration file

> Add the `--installer-config` flag to the `kommander install` command to use a custom configuration file. This same command can also be ran after the initial installation to reconfigure any applications.

Reviewer: ```suggestion
Add the `--installer-config` flag to the `kommander install` command to use a custom configuration file. To reconfigure applications, you can also run this command after the initial installation.
```

Author: resolved in a2389daa9fd58b84770bb55c6ee73e7b4e829e54
